### ISSUE TITLE
The In-App Messaging pop-up has been shown multiple times a day with 1 campaign

### ISSUE THREAD

**Issue by MichaleTuan** [firebase/firebase-ios-sdk#7704]
### [REQUIRED] Step 1: Describe your environment
  * Xcode version: 12.4
  * Firebase SDK version: FirebaseInAppMessaging v0.24.0
  * Installation method: `CocoaPods`
  * Firebase Component: inappmessaging

### [REQUIRED] Step 2: Describe the problem
I'm using Firebase In-App Messaging. When I create a new campaign with my custom event and choose `Display no more than one message every day` option. But the In-App Messaging has been shown multiple times.

#### Steps to reproduce:
Step 1: Send event `view_homescreen` from the App side when opening the app
Step 2: Go to Firebase In-App Messaging and create a new campaign
Step 3: [Style and Content] Choose modal type & fill the information
Step 4: [Target] fill in information
Step 5: [Scheduling] Choose `view_homescreen` event & `Display no more than one message every day`
Step 6: Click Review and then Publish
Step 7: Open the app -> In-App Messaging will display in Home Screen
Step 8: Closed app and open app again -> In-App Messaging will display in Home Screen again

Images: https://drive.google.com/drive/folders/110Usxv42WDwLWWYtIqav_bovxqvmH-ZP?usp=sharing 

#### Relevant Code:
```
    [FIRAnalytics logEventWithName:@"view_homescreen" parameters:@{}];
```

**Comment by rizafran** [comment]
Thanks for reporting, Michale. It seems like you're using an older version of SDK (0.24.0). Could you try to update to the [latest one](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseInAppMessaging/CHANGELOG.md) and see if you'll still encounter the issue?

**Comment by MichaleTuan** [comment]
@rizafran Thanks for your advice. I updated to the latest version (7.8.0) & it still happened. If you create a campaign with option `Display no more than one message every day` . You will see the problem.

**Comment by christibbs** [comment]
Sorry for the delay. I was able to reproduce this and I'm investigating the issue.

**Comment by sbo-nemlig** [comment]
Any news on this ?

**Comment by christibbs** [comment]
@sbo-nemlig Still working on this.

**Comment by hb24324** [comment]
@christibbs Any update when do you plan to drop the fix?

**Comment by rizafran** [comment]
Sorry for the late revert, I was able to reproduce the same behavior. With this, I'll inform the engineer and see what we can do here. Thanks.

**Comment by sbo-nemlig** [comment]
Any update i could see that you had fixed something close to it https://github.com/firebase/firebase-ios-sdk/issues/8907 , but it looks like it is still not working

**Comment by rizafran** [comment]
Hi @sbo-nemlig, we're tracking this internally via b/218320271, and will follow up if we need more information. Thanks.

**Comment by sbo-nemlig** [comment]
@rizafran  looks like a workaround is to call impressionDetected manually, i don't know if this is by design, or if the SDK should call it by it self.

**Comment by rizafran** [comment]
Hi folks, I tried reproducing the issue again using the latest SDK version (8.12.1), and it seems like the issue has disappeared. Could you also try using the latest version and see if you'll still encounter the same behavior?

**Comment by rizafran** [comment]
Hey folks, it looks like this issue has been fixed on [FIAM 8.11.0](https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseInAppMessaging/CHANGELOG.md#2021-12----v8110). I'll be closing this ticket now, but feel free to reopen if the issue persists. Thanks.
